prov/gni/src/gnix_nic.c

```diff
@@ -451,25 +450,14 @@ int gnix_nic_alloc(struct gnix_fid_domain *domain,
 
 	pthread_mutex_lock(&gnix_nic_list_lock);
 
-	if (gnix_nics_per_ptag[domain->ptag] ==
-					gnix_def_max_nics_per_ptag) {
-		list_for_each(&gnix_nic_list, elem, list) {
-			if ((elem->ptag == domain->ptag) &&
-				(elem->cookie == domain->cookie)) {
-				nic = elem;
-				break;
-			}
-		}
+	if (gnix_nics_per_ptag[domain->ptag] >= gnix_max_nics_per_ptag) {
+		assert(!dlist_empty(&domain->nic_list));
 
-		/*
-		 * nic found, balance use by removing from head and
-		 * positioning at end
-		 */
```

Comment:
"nic found" doesn't make sense anymore since I replaced the search loop with taking the first entry in the domain nic list.  That left 3 lines of code so I just chose to remove the whole comment.